---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# sigtools

<!-- badges: start -->
<!-- badges: end -->

Sigtools is an R-based visualization package, designed to enable the users with limited programming experience to produce statistical plots of continuous genomic data. It consists of several statistical visualizations that provide insights regarding the behavior of a group of signals in large regions -- such as a chromosome or the whole genome -- as well as visualizing them around a specific point or short region.

## Third-Party Requirement
[bwtool](https://github.com/CRG-Barcelona/bwtool/wiki)

[bedGraphToBigWig](https://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64.v369/)




## Installation

You can install the released version of sigtools from [CRAN](https://CRAN.R-project.org) with:

``` r
install.packages("sigtools")
```

And the development version from [GitHub](https://github.com/) with:

``` r
# install.packages("devtools")
devtools::install_github("shohre73/sigtools")
```

## Input
Genomic signals are genomic coordinates associated with coverage measurements, probabilities, or scores. *multi-column bedGraph* is the primary input format for the Sigtools visual tasks.

```{r, echo = FALSE}
df_genomicSignal <- read.csv('./test_data/E003-assays_chr21_bin200_bwtool.mcBedGraph', header = TRUE, sep = '\t')
df_genomicSignal[53846:53850,]
```


Users can convert several bedGraph files to a multi-column bedGraph file at the desired resolution by sing Sigtools `toMCbedGraph`.

## Example
Here are some examples of Sigtools application:
```{r eval=FALSE}
library(sigtools)
```

### Distribution
SigTools generates several distribution plots such as Kernel Density Distribution and Empirical Cumulative Distribution. These plots assist in estimating data's primary characteristics; namely the existence of multiple local maxima, the overall range, outliers.

Options for distribution plots are `nozero`, `enrichment` and `percentile`.

```{r eval=FALSE}
distribution("./test_data/E003-assays_chr21_bin200_bwtool.mcBedGraph", 
             outdir = "distribution", 
             plots = c("ecdf"), 
             header = TRUE, 
             percentile = 99, 
             enrichment = FALSE, 
             nozero = FALSE, 
             x_label = "", 
             y_label = "")

distribution("./test_data/E003-assays_chr21_bin200_bwtool.mcBedGraph", 
             outdir = "distribution", 
             plots = c("curve"), 
             header = TRUE, 
             percentile = 99, 
             enrichment = FALSE, 
             nozero = TRUE, 
             x_label = "", 
             y_label = "")
```

```{r, echo = FALSE, out.width="47%"}
#myimages<-list.files("images/", pattern = ".png", full.names = TRUE)
myimages <- c('./man/figures/assays_ecdf.png',
              './man/figures/assays_curve.png')
knitr::include_graphics(myimages)
```

### Autocorrelation

The autocorrelation plot displays the dependency of consecutive bins for each signal by recurrently calculating the correlation of a signal and itself when shifted.

```{r eval=FALSE}
autocorrelation("test_data/E003-assays_chr21_bin200_bwtool.mcBedGraph", 
                outdir = "autocorrelation", 
                header = TRUE, 
                resolution = 200L, 
                lag = 50L)
```

```{r, echo = FALSE, out.width="47%"}
#myimages<-list.files("images/", pattern = ".png", full.names = TRUE)
myimages <- c('./man/figures/assays_autoc.png')
knitr::include_graphics(myimages)
```

### Correlation
The `correlation` gives you a heatmap of Pearson correlation for 2 sets of signals. For better comprehension, the association of variables is encoded in both color and area.
```{r eval=FALSE}
correlation(path_mulColBedG_1 = "test_data/E003-assays_chr21_bin200_bwtool.mcBedGraph", 
            header_1 = TRUE, 
            path_mulColBedG_2 = "test_data/E003-assays_chr21_bin200_bwtool.mcBedGraph", 
            header_2 = TRUE, 
            outdir = "correlation", 
            sep = '\t', 
            img_width = 10, 
            img_height = 10, 
            fontSize = 20)
```

```{r, echo = FALSE, out.width="47%"}
#myimages<-list.files("images/", pattern = ".png", full.names = TRUE)
myimages <- c('./man/figures/assays_cor.png')
knitr::include_graphics(myimages)
```

### Aggregation
The aggregation plot analyzes the behavior of a signal upon every occurrence of a specific element.  Here, the enriched values of H3K36me3 are depicted over gene bodies of the human 21st chromosome.

```{r eval=FALSE}
aggregation(path_mulColBedG = "./test_data/E003-assays_chr21_bin200_bwtool.mcBedGraph",
            path_reg = "./test_data/Ensembl_chr21.gene_info",
            chr = 21,
            mode = "stretches",
            header = TRUE,
            sep = '\t',
            outdir = "aggregation",
            resolution = 200L,
            neighborhood = 20L,
            enrichment = TRUE)
```

```{r, echo = FALSE, out.width="47%"}
#myimages<-list.files("images/", pattern = ".png", full.names = TRUE)
myimages <- c('./man/figures/aggregation_H3K36me3.png')
knitr::include_graphics(myimages)
```

<!--

```{r, echo = FALSE, fig.align='center', out.width="50%"}
#myimages<-list.files("images/", pattern = ".png", full.names = TRUE)
myimages <- c("./man/figures/heatmap.png")
#myimages <- c('./man/figures/distributionCurve.png', 
#              './man/figures/distributionCurve_noZero.png',
#              'man/figures/distributionCurve_noZero_98.png')
knitr::include_graphics(myimages)
```


### Aggregation
```{r eval=FALSE}
aggregation("./preprocessing/uniOn.mulColBedg", 5, "./data/enhancer.bed", 'chr21')
```

```{r, echo = FALSE, out.width="47%"}
#myimages<-list.files("images/", pattern = ".png", full.names = TRUE)
myimages <- c('./man/figures/aggregation1.png', 
              './man/figures/aggregation2.png')
knitr::include_graphics(myimages)
```



```{r echo = FALSE}
#p1 <- image_ggplot('./man/figures/distributionCurve.png')
#p1 <- ggdraw() + draw_image('./man/figures/distributionCurve.png', scale = 0.9)
#p2 <- ggdraw() + draw_image('./man/figures/distributionCurve_noZero.png', scale = 0.9)

#knitr::include_graphics(c('./man/figures/distributionCurve.png', './man/figures/distributionCurve.png'), dpi = 100)

#grid.arrange('./man/figures/distributionCurve.png', './man/figures/distributionCurve.png', ncol = 2)

#library(png)
#library(grid)
#img <- readPNG('./man/figures/distributionCurve.png')
#grid.raster(img, img)
```

-->





